generate-errors/src/lib.rs

```diff
@@ -1,114 +1,213 @@
+"#;
+
+    fs::write(errors_folder_path.join("_index.md"), section_content)?;
+    fs::write(
+        errors_folder_path.join("introduction.md"),
+        introduction_content,
+    )?;
 
     Ok(())
 }
 
-pub fn parse_errors(errors_dir: &str) -> io::Result<Section> {
-    let mut errors_root_section = Section {
-        name: "Errors".to_string(),
-        content: vec![],
-        template: Some("errors.html".to_string()),
-        header: Some("Errors".to_string()),
-        order: None,
-        sort_order_reversed: false,
-    };
-    visit_dirs(
-        PathBuf::from_str(errors_dir).unwrap(),
-        &mut errors_root_section,
-    )?;
-    Ok(errors_root_section)
+pub fn write_pages(output_path: &Path, pages: HashMap<String, String>) -> anyhow::Result<()> {
+    let errors_folder_path = output_path.join("errors");
+    // Make sure the output folder exists
+    fs::create_dir_all(&errors_folder_path)?;
+
+    // Make the keys ordered so that
+    // we know the weights for the pages
+    let mut keys: Vec<&String> = pages.keys().collect();
+    keys.sort_unstable();
+
+    for (index, key) in keys.iter().enumerate() {
+        let page_content = format!(
+            r#"+++
+title = "{}"
+[extra]
+weight = {}
++++
+
+{}"#,
+            key.strip_suffix(".md").unwrap_or(key),
+            // Since the introduction page takes the
+            // zeroth position we need to treat the keys
+            // like they're one indexed to not have
+            // conflicting weights which have
+            // undefined behavior.
+            index + 1,
+            pages.get(*key).ok_or(anyhow!(
+                "The page content for {key} doesn't exist in the HashMap!"
+            ))?
+        );
+
+        fs::write(errors_folder_path.join(key.to_lowercase()), page_content)?;
+    }
+
+    Ok(())
 }
 
 #[cfg(test)]
-mod test_regex {
-    use regex::Regex;
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_error_page_content() {
+        let error_content: anyhow::Result<HashMap<String, String>> =
+            get_error_pages(Path::new("./bevy/errors"));
+        assert!(error_content.is_ok());
+    }
 
     #[test]
-    fn error_code_pattern() {
-        let error_code_pattern = Regex::new(r"B[0-9]{4}\.md").unwrap();
-        assert!(error_code_pattern.is_match("B0000.md"));
-        assert!(error_code_pattern.is_match("B9999.md"));
-
-        assert!(!error_code_pattern.is_match("E0000.md"));
-        assert!(!error_code_pattern.is_match("00000.md"));
-        assert!(!error_code_pattern.is_match("B0000.txt"));
-        assert!(!error_code_pattern.is_match("Cargo.toml"));
-        assert!(!error_code_pattern.is_match("README.md"));
+    fn test_write_section() {
+        // Fake content folder
+        let output_path = Path::new("content/learn");
+
+        let result: anyhow::Result<()> = write_section(output_path);
+        assert!(result.is_ok());
+    }
```

Comment:
Should this delete the directory afterwards?

Additionally, this and `test_write_pages` might conflict if run in parallel. I don't know if `cargo-test` runs tests in parallel by default, but it might be a possibility in the future. Instead, could each of these tests create their own custom folder so they don't interfere with one another?

There are also some crates for doing filesystem tests, which may be a more maintainable solution.

Reply:
Mm, I'll just create separate folders for either of the tests for now. Adding in crates for filesystem tests sounds like a bit much for the current problem, now if these tests relied on results of previous ones or other complex things, then I'd go for filesystem testing crates.